modules/flac/audio_stream_flac.cpp

```diff
@@ -1,219 +1,219 @@
+		if (start_buffer > 0) {
+			buffer = (buffer + start_buffer * 2);
+		}
+		int mixed = drflac_read_pcm_frames_f32(pFlac, todo, pSamples);
+		for(int i = 0; i<mixed; i++){
+			buffer[i * 2] = pSamples[i * flac_stream->channels];
+			buffer[i*2 + 1] = pSamples[i * flac_stream->channels + flac_stream->channels-1];
+		}
+
+		todo -= mixed;
+		frames_mixed += mixed;
+
+		if (todo) {
+			//end of file!
+			if (flac_stream->loop) {
+				//loop
+				seek(flac_stream->loop_offset);
+				loops++;
+				// we still have buffer to fill, start from this element in the next iteration.
+				start_buffer = p_frames - todo;
+			} else {
+				for (int i = p_frames - todo; i < p_frames; i++) {
+					p_buffer[i] = AudioFrame(0, 0);
+				}
+				active = false;
+				todo = 0;
+			}
+		}
+		AudioServer::get_singleton()->audio_data_free(pSamples);
+	}
+
+	
+	
+}
+
+float AudioStreamPlaybackFLAC::get_stream_sampling_rate() {
+
+	return flac_stream->sample_rate;
+}
+
+void AudioStreamPlaybackFLAC::start(float p_from_pos) {
+
+	active = true;
+	seek(p_from_pos);
+	loops = 0;
+	_begin_resample();
+}
+
+void AudioStreamPlaybackFLAC::stop() {
+
+	active = false;
+}
+bool AudioStreamPlaybackFLAC::is_playing() const {
+
+	return active;
+}
+
+int AudioStreamPlaybackFLAC::get_loop_count() const {
+
+	return loops;
+}
+
+float AudioStreamPlaybackFLAC::get_playback_position() const {
+	return float(frames_mixed) / float(flac_stream->sample_rate);
+}
+void AudioStreamPlaybackFLAC::seek(float p_time) {
+	if (!active)
+		return;
+
+	if (p_time >= flac_stream->get_length()) {
+		p_time = 0;
+	}
+	frames_mixed = flac_stream->sample_rate * p_time;
+	drflac_seek_to_pcm_frame(pFlac, frames_mixed);
+}
+
+AudioStreamPlaybackFLAC::~AudioStreamPlaybackFLAC() {
+	if (pFlac) {
+		//AudioServer::get_singleton()->audio_data_free(pFlac);
+		pFlac = NULL;
+	}
+}
+
+Ref<AudioStreamPlayback> AudioStreamFLAC::instance_playback() {
+	Ref<AudioStreamPlaybackFLAC> flacs;
+
+	ERR_FAIL_COND_V(data == NULL, flacs);
+
+	flacs.instance();
+	flacs->flac_stream = Ref<AudioStreamFLAC>(this);
+
+	flacs->pFlac = drflac_open_memory(data, data_len, NULL);
+
+	flacs->frames_mixed = 0;
+	flacs->active = false;
+	flacs->loops = 0;
+	
+	if (!flacs->pFlac) {
+
+		ERR_FAIL_COND_V(!flacs->pFlac, Ref<AudioStreamPlaybackFLAC>());
+	}
+
+	return flacs;
+}
+
+String AudioStreamFLAC::get_stream_name() const {
+
+	return ""; //return stream_name;
+}
+
+void AudioStreamFLAC::clear_data() {
+	if (data) {
+		AudioServer::get_singleton()->audio_data_free(data);
+		data = NULL;
+		data_len = 0;
+	}
+}
+
+void AudioStreamFLAC::set_data(const PoolVector<uint8_t> &p_data) {
+
+	int src_data_len = p_data.size();
+
+	PoolVector<uint8_t>::Read src_datar = p_data.read();
+
+	drflac *pflac = drflac_open_memory(src_datar.ptr(), src_data_len, NULL);
+
+	channels = pflac->channels;
+	sample_rate = pflac->sampleRate;
+	length = ((float(pflac->totalPCMFrameCount) * float(channels)) / float(sample_rate)) / float(channels);

```

Comment:
Multiplying, then dividing by channels seems redundant.
```suggestion
	length = float(pflac->totalPCMFrameCount) / float(sample_rate);
```

Reply:
Did some calculation here and yep, it is redundant!